### PR TITLE
feat(client/server): Adding unique fields #1493  

### PR DESCRIPTION
**Pull request by lalit8347**
Open #1493

Added ToggleField for unique option in amplication-client
Corresponding changes to graphql queries for unique option at server and client side.
Added unique option in EntityField model at server side.

### CODE REVIEW

**Review by yuval-hazaz — COMMENTED** [review]
before I continue with the review please revert the changes in the versions of the packages (lerna, and @graphql-codegen/cli).

I do not expect to see any changes in the packge.json or pakcage-lock.json as part of this PR

File: package.json [hunk 1]
```diff
@@ -20,10 +20,10 @@
     "generate-test-data-service": "lerna run --scope @amplication/data-service-generator generate-test-data-service"
   },
   "dependencies": {
-    "lerna": "3.22.1"
+    "lerna": "^4.0.0"
```
[the remark this answers is not in the document]
Comment (lalit8347) [reply package.json:23]: I might have updated it while installing packages.

File: packages/amplication-cli/src/models.ts [hunk 2]
```diff
@@ -2,6 +2,10 @@ export type Maybe<T> = T | null;
 export type Exact<T extends { [key: string]: unknown }> = {
   [K in keyof T]: T[K];
 };
+export type MakeOptional<T, K extends keyof T> = Omit<T, K> &
```
[the remark this answers is not in the document]
Comment (lalit8347) [reply packages/amplication-cli/src/models.ts:5]: Yeah I agree, I will revert it in next PR.

File: package.json [hunk 3]
```diff
@@ -20,10 +20,10 @@
...
   },
   "devDependencies": {
-    "@graphql-codegen/cli": "1.21.6",
+    "@graphql-codegen/cli": "^1.21.4",
```
[the remark this answers is not in the document]
Comment (lalit8347) [reply package.json:26]: I am not sure about how this is changed, but I will revert it back.

**Review by yuval-hazaz — COMMENTED** [review]
Well done! everything looks good.
Please remove the changes included in package-lock.josn

File: packages/amplication-data-service-generator/src/server/user-entity.ts [hunk 4]
```diff
@@ -25,6 +26,7 @@ export const USER_PASSWORD_FIELD: EntityField = {
   displayName: "Password",
   dataType: EnumDataType.Password,
   required: true,
+  unique: true,
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/user-entity.ts:29]: this should not be unique

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-data-service-generator/src/server/user-entity.ts (hunk 4 above)
Comment (lalit8347) [reply packages/amplication-data-service-generator/src/server/user-entity.ts:29]: Yes I will it change in next PR.

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**